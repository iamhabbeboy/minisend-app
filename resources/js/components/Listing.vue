<template>
    <div>
        <Filters />
        <table class="table table-striped">
            <thead>
                <tr>
                    <th scope="col">#</th>
                    <th scope="col">ID</th>
                    <th scope="col">Sender</th>
                    <th scope="col">Recipient</th>
                    <th scope="col">Subject</th>
                    <th scope="col">Content</th>
                    <th scope="col">Attachment</th>
                    <th scope="col">Status</th>
                    <th scope="col">Option</th>
                </tr>
            </thead>
            <tbody>
                <tr>
                    <td scope="row">#</td>
                    <td>1012</td>
                    <td>Abiodun Solomon</td>
                    <td>Damola</td>
                    <td>testing</td>
                    <td>sldkfdf</td>
                    <td>sldfkjsdklf</td>
                    <td>Failed</td>
                    <td>sldfkj</td>
                </tr>
            </tbody>
        </table>
    </div>
</template>
<script>
import Filters from "./Filters";
export default {
    components: {
        Filters
    }
};
</script>
