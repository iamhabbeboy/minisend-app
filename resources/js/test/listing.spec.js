import { mount, shallowMount, createLocalVue } from "@vue/test-utils";
import Listing from "../components/Listing.vue";
import Vuex from "vuex";

const localVue = createLocalVue();
let wrapper;
localVue.use(Vuex);
describe("Listing component Listing.vue", () => {
    let actions;
    let state;
    let store;

    beforeEach(() => {
        actions = {
            getEmails: jest.fn()
        };
        state = {
            emails: jest.fn()
        };

        store = new Vuex.Store({
            state,
            actions
            // getters
        });

        wrapper = shallowMount(Listing, {
            store,
            localVue,
            propsData: {
                emails: [],
                display: ""
            },
            mocks: {
                pagination: []
            },
            stubs: ["router-view", "router-link"],
            store,
            localVue
        });
    });

    test("component snapshot", () => {
        expect(wrapper).toMatchSnapshot();
    });

    test("component is a Vue instance", () => {
        expect(wrapper.isVueInstance).toBeTruthy();
    });

    test("render next button for pagination", () => {
        const button = wrapper.find(".btn-info");
        expect(button.text()).toBe("Next");
    });

    test("render previous button for pagination", () => {
        const button = wrapper.find(".btn-default");
        expect(button.text()).toBe("Previous");
    });
});
