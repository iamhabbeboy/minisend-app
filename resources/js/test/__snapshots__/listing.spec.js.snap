// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Listing component Listing.vue component snapshot 1`] = `
VueWrapper {
  "_emitted": Object {},
  "_emittedByOrder": Array [],
  "isFunctionalComponent": undefined,
}
`;
